psyneulink/core/components/functions/nonstateful/combinationfunctions.py

```diff
@@ -727,19 +728,25 @@ class Parameters(CombinationFunction.Parameters):
         changes_shape = Parameter(True, stateful=False, loggable=False, pnl_internal=True)
 
     @check_user_specified
-    @tc.typecheck
+    @beartype
     def __init__(self,
-                 # weights: tc.optional(parameter_spec)=None,
-                 # exponents: tc.optional(parameter_spec)=None,
+                 # weights:  Optional[ValidParamSpecType] = None,
+                 # exponents:  Optional[ValidParamSpecType] = None,
                  weights=None,
                  exponents=None,
                  default_variable=None,
-                 operation: tc.optional(tc.enum(SUM, PRODUCT)) = None,
-                 scale: tc.optional(parameter_spec) = None,
-                 offset: tc.optional(parameter_spec) = None,
+                 operation: Optional[Literal['sum', 'product']] = None,
+                 scale: Optional[ValidParamSpecType] = None,
+                 offset: Optional[ValidParamSpecType] = None,
                  params=None,
                  owner=None,
-                 prefs: tc.optional(is_pref_set) = None):
+                 prefs:  Optional[ValidPrefSet] = None):
+
+        if scale is not None and not np.isscalar(scale):
+            raise ValueError("scale must be a scalar")
```

Comment:
If desired, this can be replaced with a method like this (name important) on `Reduce.Parameters`:

```python
        def _validate_scale(self, scale):
            if not np.isscalar(scale):
                return "scale must be a scalar"
```
which would also support subclasses and validation at post-construction assignments